File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/houses/HouseBuyRequestMessage.scala





















// Generated on 05/03/2013 10:34:43
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait HouseBuyRequestMessage extends MessageDeserializer {
  val messageId = 5738

  protected trait Message extends BaseMessage with Serializable {
    val messageId = HouseBuyRequestMessage.messageId

    val proposedPrice: Int

    def serialize(buf: Buffer) {
      buf.writeInt(proposedPrice)
    }

  }

  protected def create(proposedPrice: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val proposedPrice = buf.readInt()

    create(proposedPrice)
  }

}

object HouseBuyRequestMessage extends HouseBuyRequestMessage {
  case class MessageImpl(proposedPrice: Int) extends Message
  type Target = MessageImpl

  override def create(proposedPrice: Int) = MessageImpl(proposedPrice)

  def apply(proposedPrice: Int) = create(proposedPrice)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/HouseBuyRequestMessage.java


// Generated on 05/08/2013 19:37:46
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class HouseBuyRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5738;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int proposedPrice;
    
    public HouseBuyRequestMessage() { }
    
    public HouseBuyRequestMessage(int proposedPrice) {
        this.proposedPrice = proposedPrice;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(proposedPrice);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        proposedPrice = buf.readInt();
        if (proposedPrice < 0)
            throw new RuntimeException("Forbidden value on proposedPrice = " + proposedPrice + ", it doesn't respect the following condition : proposedPrice < 0");
    }
    
}
